dbt_meshify/dbt_projects.py

```diff
@@ -184,6 +168,24 @@ def split(
 
         return subproject
 
+    def add_model_contract(self, unique_id: str) -> None:
+        """Adds a model contract to the model's yaml"""
+        
+        # get the patch path for the model
+        node = self.get_manifest_node(unique_id)
+        yml_path = node.patch_path.split("://")[1] if node.patch_path else None
+        # if the model doesn't have a patch path, create a new yml file in the models directory
+        # TODO - should we check if there's a model yml file in the models directory and append to it?
+        if not yml_path:
+            yml_path = "/".join(node.original_file_path.split(".")[0].split("/")[:-1]) + "_models.yml"
```

Comment:
Very clever way to parse the file name without the extension.

Another approach for this would be to leverage the `pathlib` library's stem property.

https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem

Reply:
totally aligned, i think there's a lot of cleanup to be done on the file manager class with pathlib -- opened #27 to reflect that!